workshops/rust_text_game/README.md

```diff
@@ -0,0 +1,405 @@
+
+Let's get started!
+
+### Part 4.1: Output some text
+
+So right now, our code looks like this.
+```rust
+fn main() {
+  println!("Hello, World!");
+}
+```
+
+ But `Hello, World!` doesn't feel like something you'd see in a game, right?. So, let's change that. Change the text inside the quotes to something else. We'll change it to `Hello brave explorer! Want to embark on an adventure?`.
+ 
+Now run your program again. Instead of `Hello, World`, you should see what you wrote above
+
+### Part 4.2: User input
+
+A game needs to give the ability the option to interact with it and affect its course.
+
+... After all, [it's what makes a Subaru a Subaru!](https://www.youtube.com/watch?v=aYVoAioo8Ww)
+
+So, we'll write a new function that will accept read the player's answer and react accordingly.
+
+For purposes of simplicity, the player will only be supposed to answer with a word starting with "Y", which includes words such as 'Yes', 'Yep', 'Yeah', or 'Yellow' (as a funny unintended consequence), or, likewise, "N".
+
+`src/prompts.rs`:
+```rust
+use std::io;
+
+fn prompt() -> bool {
+    let mut input = String::new();
+    io::stdin().read_line(&mut input).unwrap();
+
+    return input.to_ascii_lowercase().starts_with("y");
+}
+```
+
+Now, I know that this is a lot to take in, but don't fret! Let's just go through what's happening here once more, line-by-line:
+
+   - `use std::io;`
+    
+    Importing libraries allows us to use code either installed by cargo, or bundled with Rust (standard library). By importing `std::io` (where `std` means standard library aka included with Rust), we gain access to more functions that allow us to perform "input/output" operations (like `println!()`, but more advanced!).
+
+- `fn prompt() -> bool {`
+
+    All functions are meant to return some output, which is either visible to the player, to other programs or parts of the same program. In this case, we define a new function called `prompt()`, which will return a value known as a `bool`, which stands for "boolean". Boolean data types can only accept two values, which are denoted with the words `true` or `false` in the Rust programming language. They're named after the mathematician George Boole and are possibly the simplest instance of [Boolean algebra](https://en.wikipedia.org/wiki/Boolean_algebra) in the field of computer science.
+
+- `let mut input = String::new();`
+
+    Here, we create a *mutable* `String`. *Mutability* defines whether you can change the value of a variable. Since we're *initializing* a new string (which means that we declare that the variable `input` exists and is a `String`, but there haven't been any values that have been *declared* to it yet.)
+
+- `io::stdin().read_line(&mut input).unwrap();`
+
+    Here, using the `io` library again, we capture whatever the player is typing until they hit `Enter` inside the `input` variable.
```

Comment:
```suggestion
    Here, using the std's `io` module again, we capture whatever the player is typing until they hit `Enter` inside the `input` variable. The `&mut input` you see is Rust's syntax for creating a mutable _reference_ to a variable. In this case, that's how the `read_line` method is able to read from your standard input buffer; it places the bytes into the memory referenced by the mutable reference that it is passed. In this case, you're passing a mutable reference to `input`, so the resulting line will be placed _directly_ in that `String` without any new `String`s being created. Note that we explicitly had to add the `mut` specifier to our reference; Rust references and variables are instantiated as immutable unless we specify otherwise. In our case, we are specifying it as mutable so that it can be, well, mutated by the `read_line` method
```